### PR TITLE
Closes #5175: Add Glean keys for distinguishing between fatal and non-fatal native crashes

### LINKED ISSUE #5175: Crash reporter: Allow to distinguish between main process and other process crashes.
To measure stability and compare with Fennec we want to know whether a crash happened in the main process (and therefore takes down the whole app) or whether it happened in another process (e.g. the content process).

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by rocketsroger** [comment]
bors r=travis79

**Comment by travis79** [comment]
> This requires data review.

Good catch!

**Comment by rocketsroger** [comment]
Request for data collection review
---
1. What questions will you answer with this data?
            This data will tell us if a native code crash was fatal or non-fatal.
 
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
            Yes, this gives us an understanding of user experience since a fatal crash is much more disruptive to the user.
 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
            No alternative is available to answer this question.
 
4. Can current instrumentation answer these questions?
            No, current instrumentation only tells us there is a native code crash.
 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement   Description | Data Collection   Category | Tracking Bug #
-- | -- | --
Crash is   fatal/non-fatal | category 1 - technical   data | https://github.com/mozilla-mobile/android-components/issues/5175

6. How long will this data be collected? Choose one of the following:
            I want to permanently monitor this data. (Roger Yang)
 
7. What populations will you measure?
            No filters - all locales, channels, etc.
 
8. If this data collection is default on, what is the opt-out mechanism for users?
            Default Glean SDK opt-out mechanism.
 
9. Please provide a general description of how you will analyze this data.
            When high amount of fatal crashes occurs, we will look at crash reporting to see if there are any recently introduced crashes.
 
10. Where do you intend to share the results of your analysis?
            Internally in the fenix/android-components teams.
 
11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
            No third-party tool will use this data.

**Comment by rocketsroger** [comment]
@liuche @boek Please review the request for data collection.  Thanks!

File: components/lib/crash/metrics.yaml [hunk 1]
```diff
@@ -18,7 +18,8 @@ crash_metrics:
     labels:
       - uncaught_exception
       - caught_exception
-      - native_code_crash
+      - fatal_native_code_crash
+      - nonfatal_native_code_crash
     bugs:
       - https://bugzilla.mozilla.org/1553935
     data_reviews:
```
Comment (liuche) [inline components/lib/crash/metrics.yaml]: Nit: also add a reference to this data review

Comment (rocketsroger) [reply components/lib/crash/metrics.yaml]: Understood.  Thanks!

File: components/lib/crash/metrics.yaml [hunk 2]
```diff
@@ -18,11 +18,14 @@ crash_metrics:
     labels:
       - uncaught_exception
       - caught_exception
-      - native_code_crash
+      - fatal_native_code_crash
```
Comment (Dexterp37) [inline components/lib/crash/metrics.yaml]: @mdboom @fbertsch is Glean allowed to rename labels?

Comment (mdboom) [reply components/lib/crash/metrics.yaml]: It's allowed -- they are transmitted and stored as strings.

However, I would recommend documenting in the description the purpose of the one being removed *if* we ever expect analysts to look at the old data containing that label.

**Comment by rocketsroger** [comment]
> r+! Please consider whether or not adding documentation for the label being removed is of any use for your case.

Thanks! Agreed, I'll add the documentation.

File: components/lib/crash/metrics.yaml [hunk 3]
```diff
@@ -15,14 +15,19 @@ crash_metrics:
       Counts the number of crashes that occur in the application.  This measures only the counts of
       each crash in association with the labeled type of the crash. The labels correspond to the
       types of crashes handled by lib-crash.
+
+      Deprecated: native_code_crash replaced by fatal_native_code_crash and nonfatal_native_code_crash
```
Comment (Dexterp37) [inline components/lib/crash/metrics.yaml]:
```suggestion
      Deprecated: `native_code_crash` replaced by `fatal_native_code_crash` and `nonfatal_native_code_crash`
```

File: components/lib/crash/docs/metrics.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ The following metrics are added to the ping:
 
 | Name | Type | Description | Data reviews | Extras | Expiration |
 | --- | --- | --- | --- | --- | --- |
-| crash_metrics.crash_count |[labeled_counter](https://mozilla.github.io/glean/book/user/metrics/labeled_counters.html) |Counts the number of crashes that occur in the application.  This measures only the counts of each crash in association with the labeled type of the crash. The labels correspond to the types of crashes handled by lib-crash.  |[1](https://bugzilla.mozilla.org/show_bug.cgi?id=1553935#c3)|<ul><li>uncaught_exception</li><li>caught_exception</li><li>native_code_crash</li></ul>|never |
+| crash_metrics.crash_count |[labeled_counter](https://mozilla.github.io/glean/book/user/metrics/labeled_counters.html) |Counts the number of crashes that occur in the application.  This measures only the counts of each crash in association with the labeled type of the crash. The labels correspond to the types of crashes handled by lib-crash.  |[1](https://bugzilla.mozilla.org/show_bug.cgi?id=1553935#c3)|<ul><li>uncaught_exception</li><li>caught_exception</li><li>fatal_native_code_crash</li><li>nonfatal_native_code_crash</li></ul>|never |
```
Comment (Dexterp37) [inline components/lib/crash/docs/metrics.md]: Looks like the docs where not updated: any chance you could build so that this file gets updated?

Comment (rocketsroger) [reply components/lib/crash/docs/metrics.md]: Sure, thanks for catching this.

**Comment by rocketsroger** [comment]
bors r=Dexterp37,  travis79

**Comment by rocketsroger** [comment]
bors retry

**Pull request merged**